### PR TITLE
Add watcher on Linux: change fs to node-watch

### PR DESCRIPTION
**Pull request by Naerriel**
## Description
Fix to #13140.
It allows watcher to detect changes in files on Linux OS, as recursive option in `fs.watch` isn't supported on Linux.
It adds a new developer dependency: [node-watch](https://www.npmjs.com/package/node-watch).

## How has this been tested?

I modified a file and the watcher recognized it and built the project.

My testing environment is Ubuntu 16.04.4

## Types of changes
Bug fix

## Checklist:
- [X] My code is tested.
- [X] My code follows the WordPress code style. 
- [X] My code follows the accessibility standards. 
- [X] My code has proper inline documentation. 
- [X] I've included developer documentation if appropriate.

### CODE REVIEW

**Comment by youknowriad** [comment]
Any linux users to review this, @nosolosw maybe?

**Comment by nosolosw** [comment]
Adding myself as a reviewer, will come to this later in the day.

**Comment by nosolosw** [comment]
cc @notnownikki  as well

**Comment by notnownikki** [comment]
:grinning: :grinning: :grinning: I no longer have to rebuild everything from scratch when making changes! :dancing_women: 

Yes, works. Lovely, delightful, wonderful!

**Comment by gziolo** [comment]
It would be nice to test with Windows, too :)

File: package.json [hunk 1]
```diff
@@ -96,7 +96,8 @@
 		"lint-staged": "7.3.0",
 		"lodash": "4.17.10",
 		"mkdirp": "0.5.1",
-		"node-sass": "4.11.0",
+	  	"node-sass": "4.11.0",
```
Comment (aduth) [inline package.json]: There's mixed whitespace here. Did you add these manually? Running `npm install` reveals there are changes introduced to the branch.

**Review by aduth — CHANGES_REQUESTED** [review]
Can you re-run `npm install` with the latest version of npm (v6.7.0) ? For me, many changes are introduced.

(Related: #13462)

**Comment by aduth** [comment]
Tested macOS:

image

Tested Windows 10:

image

File: package.json (hunk 1 above)
Comment (Naerriel) [reply package.json]: Yes, I did add those manually, as the indentation haven't had the same length. I will fix it straight away.

**Comment by Naerriel** [comment]
@aduth I've ran into a trouble regarding auto addition of `"optional": true` parameters to `package-lock.json` which happens probably because I'm on Linux and not on Mac but now everything should be in order. Can you check it, please?

**Comment by aduth** [comment]
Ah yes, there are some platform-specific issues with npm on Linux, where `optional` flags will differ.

I've pushed up the resulting changes I see on my end with `npm install` in e6b1a22.

**Review by aduth — APPROVED** [review]
Thanks, nice improvement 👍

**Comment by Naerriel** [comment]
Thank you. Yes, that was my original code but I deleted the `optional` flags manually, thinking they are unintended.

Thank you :smile:

**Comment by notnownikki** [comment]
@Naerriel and @aduth you're my heroes of the month!

**Comment by Rahmon** [comment]
Thanks for fix. I've spent hours to find out the problem.

**Comment by Naerriel** [comment]
You're welcome. It would be impossible if you haven't diagnosed it first :smile:

**Pull request merged**